### PR TITLE
Setup readthedocs sphinx

### PR DESCRIPTION
**Pull request by hakonanes**
This addresses #108 (might do multiple PRs before closing the issue).

* Add `doc` alongside `test` to the extra `dev` requirements in setup.py, with `sphinx` and `shinx-rtd-theme`. Min. versions chosen based on what we use in `kikuchipy` (it works)
* Link resolve nicked from SciPy's read the docs (rtd) config file (credited in docstring). This will make sure that "source" links in the API directs to relevant lines in GitHub, and not just code via rtd (try it out by clicking "source" here https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.fft2.html#scipy.fft.fft2)
* Full author list in footer (should this be changed to just "orix developers"?)

Todo:
- [x] Create full one-page API reference in reference.rst
- [x] Decide whether we should use the pyxem logo used on GitHub as a logo at orix rtd, and perhaps as a favicon?
- [x] Add pyxem/orix logo to top of sidebar and as favicon
- [x] Decide on any other pages in addition to the API, like "Installation", "Related projects", "Contributing", "Changelog", "Cite", etc.
- [x] Add a "docs" badge to README.rst
- [x] Merge #114 into this branch
- [x] Fix up docstrings and function/class references

### CODE REVIEW

**Comment by hakonanes** [comment]
Unless I push this branch to the repo, I cannot test how it looks on https://orix.rtfd.io... It looks fine locally, but would be nice for others to have a look at this PR via that link as well. @dnjohnstone @pc494 are you OK with me pushing this branch to this repo?

**Comment by dnjohnstone** [comment]
> Unless I push this branch to the repo, I cannot test how it looks on https://orix.rtfd.io... It looks fine locally, but would be nice for others to have a look at this PR via that link as well. @dnjohnstone @pc494 are you OK with me pushing this branch to this repo?

Yep, go for it.

**Comment by hakonanes** [comment]
Should master into this branch after #114 is potentially merged into master, before merging this into master.

**Comment by dnjohnstone** [comment]
Is this thing good to go now?

**Comment by hakonanes** [comment]
> Is this thing good to go now?

Waiting for the orix logo from @pc494. After that I'll just have a look over and ping for a review.

**Comment by pc494** [comment]
Sorry, mea culpa, dealt with that now.

**Comment by hakonanes** [comment]
I think reviewing is best done by looking here https://orix.readthedocs.io/en/latest/

**Review by dnjohnstone — APPROVED** [review]
This looks good to me - thanks very much @hakonanes for taking the effort to do this! Will merge when the tests pass.

File: doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+Related, open-source projects that users of orix might find useful:
+
+- `pyxem <https://github.com/pyxem/pyxem>`_: Python library for multi-dimensional
+  diffraction microscopy.
+- `diffsims <https://github.com/pyxem/diffsims>`_: Python library for simulating
+  diffraction.
+- `kikuchipy <https://kikuchipy.org>`_: Python library for processing and analysis of
+  electron backscatter diffraction (EBSD) patterns.
+- `MTEX <https://mtex-toolbos.github.io>`_: Matlab toolbox for analyzing and
```
Comment (pc494) [inline doc/index.rst:32]: `mtex-toolbox.github.io`

**Comment by hakonanes** [comment]
Note that I've added a docs-badge to the README!

**Review by pc494 — APPROVED** [review]
Found 1 typo in a link, but other than it looks really good. Thanks @hakonanes

**Comment by hakonanes** [comment]
Two things:
* So far, we have one version up, `latest`, built from the `master` branch. When we release 0.5, I will set up a `stable` that builds from the release tag (like 0.5.x, I do this for kikuchipy)
* If you would like maintainer access to orix.readthedocs.io, you'll need to create a user and I'll add you

**Comment by pc494** [comment]
>    So far, we have one version up, `latest`, built from the `master` branch. When we release 0.5, I will set up a `stable` that builds from the release tag (like 0.5.x, I do this for kikuchipy)
 
Sounds good

> If you would like maintainer access to orix.readthedocs.io, you'll need to create a user and I'll add you

Yes please, pc494 connected to this github account

**Comment by hakonanes** [comment]
> Yes please, pc494 connected to this github account

done!

**Pull request merged**